src/websockets/client.py

```diff
@@ -364,26 +372,26 @@ class Connect:
 
     def __init__(
         self,
-        uri,
+        uri: str,
         *,
-        create_protocol=None,
-        ping_interval=20,
-        ping_timeout=20,
-        close_timeout=None,
-        max_size=2 ** 20,
-        max_queue=2 ** 5,
-        read_limit=2 ** 16,
-        write_limit=2 ** 16,
-        loop=None,
-        legacy_recv=False,
-        klass=WebSocketClientProtocol,
-        timeout=10,
-        compression="deflate",
-        origin=None,
-        extensions=None,
-        subprotocols=None,
-        extra_headers=None,
-        **kwds
+        create_protocol: Optional[Type[WebSocketClientProtocol]] = None,
+        ping_interval: float = 20,
+        ping_timeout: float = 20,
+        close_timeout: Optional[float] = None,
+        max_size: int = 2 ** 20,
+        max_queue: int = 2 ** 5,
+        read_limit: int = 2 ** 16,
+        write_limit: int = 2 ** 16,
+        loop: Optional[asyncio.AbstractEventLoop] = None,
+        legacy_recv: bool = False,
+        klass: Type[WebSocketClientProtocol] = WebSocketClientProtocol,
```

Comment:
Suggestion to use a [type variable with an upper bound](https://mypy.readthedocs.io/en/latest/generics.html?highlight=typevar#type-variables-with-upper-bounds):

Somewhere above...

`WSCP = TypeVar('WSCP', bound=WebSocketClientProtocol)`

... then, in the signature...

`klass: Type[WSCP] = WebSocketClientProtocol`

... and return `WSCP` instead of `WebSocketClientProtocol` in methods below.

This allows to return the actual type of the instance being created instead of the upper bound and prevents the user from needing to `cast`.

Reply:
I tried that; it causes errors I don't understand well and requests additional type annotations:

```
src/websockets/server.py:805: error: Incompatible default for argument "klass" (default has type "Type[WebSocketServerProtocol]", argument has type "Type[WebSocketServerProtocolType]")
src/websockets/server.py:865: error: Argument 1 to "WebSocketServerProtocol" has incompatible type "Callable[[WebSocketServerProtocolType, str], Awaitable[Any]]"; expected "Callable[[WebSocketServerProtocol, str], Awaitable[Any]]"
src/websockets/client.py:370: error: Incompatible default for argument "klass" (default has type "Type[WebSocketClientProtocol]", argument has type "Type[WebSocketClientProtocolType]")
src/websockets/client.py:509: error: Need type annotation for 'protocol'
src/websockets/client.py:534: warning: Returning Any from function declared to return "WebSocketClientProtocolType"
src/websockets/__main__.py:98: error: Need type annotation for 'websocket'
```

Not sure how to proceed from there.